js/gsce-custom.js

```diff
@@ -0,0 +1,176 @@
+const DOMAIN_URL_LENGTH = "https://mechanicalrock.github.io/".length;
+const MONTHS = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+const VISIBLE_PAGE_NUMBERS = 10;
+window.__gcse = {
+    searchCallbacks: {
+      web: {
+        rendered: hidePosts,
+      },
+    },
+  };
+var homePagePostsElement;
+
+window.onload = function() {
+   var clearButtonExists = setInterval(() => {
+        var clearButtonElements = document.getElementsByClassName("gsst_a");
+        if(clearButtonElements.length > 0)
+        {
+            clearButtonElements[0].addEventListener("click", showPosts);
+            clearInterval(clearButtonExists);  
+        }
+    }, 100);
+
+    var homePagePostsElements = document.getElementsByClassName("home-page-posts");
+    if(homePagePostsElements.length == 0)
+    {
+        console.error("Home page posts element not found.");
+        return;
+    }
+    homePagePostsElement = homePagePostsElements[0];
+
+    setupPagination();
+}
+
+function hidePosts()
+{
+    var resultsElements = document.getElementsByClassName("gsc-results-wrapper-nooverlay");
+    if(resultsElements.length == 0)
+    {
+        console.error("Results element not found.");
+        return;
+    }
+
+    var resultsElement =  resultsElements[0]
+    resultsElement.style.display = "block";
+    homePagePostsElement.style.display = "none";
+
+    getDateFromLink();
+}
+
+function showPosts()
+{
+    var resultsElements = document.getElementsByClassName("gsc-results-wrapper-nooverlay");
+    if(resultsElements.length == 0)
+    {
+        console.error("Results element not found.");
+        return;
+    }
+
+    var resultsElement =  resultsElements[0];
+    resultsElement.style.display = "none";
+    homePagePostsElement.style.display = "block";
+}
+
+/*
+    Used to attach date above the post link in google results 
+*/
+function getDateFromLink()
+{
+    var linksExist = setInterval(() => {
+        var links = document.getElementsByClassName("gs-per-result-labels");
+        links = [...links]
+        if(links.length > 0)
+        {
+            
+            links.forEach(link => {
+                var url = link.getAttribute("url");
+                if(!url)
+                {
+                    return;
+                }
+                
+                var dateString = url.substr(DOMAIN_URL_LENGTH, 10);
+                var date = new Date(dateString);
+                var webResult = link.parentElement.parentElement.parentElement;
```

Comment:
What confidence do we have with the parentElement traversing occuring at the 3 levels?

Reply:
How else could we compute it?

Reply:
Unsure if there is another way to compute this

Reply:
@Thashin - last outstanding comment:

Can this be a CSS Selector? If the HTML structure changes, this would likely break.

e.g. can we add an id or name attribute to the target element?

Reply:
Fixed